### PR TITLE
Replace checked-in ZIPs with dynamic dependencies

### PR DESCRIPTION
**Pull request by downsrob**
Signed-off-by: Clay Downs <downsrob@amazon.com>

*Issue #, if available:*
https://github.com/opensearch-project/index-management/issues/305

*Description of changes:*
Replace checked-in ZIPs with dynamic dependencies. The current job scheduler zip used for integration tests is downloaded from ci.opensearch.org. The backwards compatibility test job scheduler and index management versions are downloaded from https://github.com/opendistro-for-elasticsearch artifacts. These dependencies are downloaded during the gradle configuration phase, and will be refreshed with any task run.

Removes the mustache zip entirely, as I couldn't find any reference to it being used.

*CheckList:*
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -234,6 +240,17 @@ test {
...
+ext.getPluginResource = { download_to_folder, download_from_src ->
+    project.mkdir download_to_folder
+    ant.get(src: download_from_src,
+            dest: download_to_folder,
+            httpusecaches: false)
+    return fileTree(download_to_folder).getSingleFile()
+}
+
+// Download the job scheduler test dependency
+getPluginResource(job_scheduler_resource_folder, job_scheduler_build_download)
```
Comment (thalurur) [inline build.gradle:252]: by calling this at this level in build.gradle mean this will get executed always for all targets?

Comment (downsrob) [reply build.gradle:252]: That's right. I believe that we also initialize the integTest clusters with the plugin zips for all targets in the configuration phase, so to get around downloading the resource always, we would need to refactor the cluster setup, which I didn't have time to take on at this point

Comment (thalurur) [reply build.gradle:252]: got it, lets create an issue for this - we don't need to address it immediately

**Review by thalurur — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (downsrob) [reply build.gradle:252]: https://github.com/opensearch-project/index-management/issues/332

**Pull request merged**